src/main/java/seedu/clinic/logic/commands/EditCommand.java

```diff
@@ -305,7 +330,13 @@ public void setRemark(Remark remark) {
          * A defensive copy of {@code products} is used internally.
          */
         public void setProducts(Set<Product> products) {
-            this.products = (products != null) ? new HashSet<>(products) : null;
+            if (products == null) {
+                this.products = null;
+            } else if (products.isEmpty()) {
+                this.products = null;
```

Comment:
Perhaps we could combine this into one if-condition!

Reply:
I think if combining the two will result in NullPointerException since if products == null, it won't have isEmpty method.